crates/services/gas_price_service/src/v1/da_source_adapter/block_committer_costs.rs

```diff
@@ -6,57 +9,334 @@ use crate::v1::da_source_adapter::{
     DaBlockCosts,
 };
 use anyhow::anyhow;
-use reqwest::Url;
+use fuel_core_types::blockchain::primitives::DaBlockHeight;
 use serde::{
     Deserialize,
     Serialize,
 };
 
-/// This struct is used to denote the block committer da gas price source,,
+#[async_trait::async_trait]
+trait BlockCommitterApi: Send + Sync {
+    // Used on first run to get the latest costs and seqno
+    async fn get_latest_costs(&self) -> DaBlockCostsResult<Option<RawDaBlockCosts>>;
+    // Used to get the costs for a specific seqno
+    async fn get_costs_by_seqno(
+        &self,
+        number: u64,
+    ) -> DaBlockCostsResult<Option<RawDaBlockCosts>>;
+    async fn get_bundles_by_range(
+        &self,
+        range: core::ops::Range<u64>,
+    ) -> DaBlockCostsResult<Vec<Option<RawDaBlockCosts>>>;
```

Comment:
its an inclusive range, so I assume we should actually increment by 1. We aren't using this yet, but it will be a part of a more comprehensive solution.

Reply:
addressed docstring in https://github.com/FuelLabs/fuel-core/pull/2265/commits/5330975014afec7ed0aad7401c701d4a3921f8fd

Reply:
> its an inclusive range

I'm not sure if this answered my question. The committer doesn't have data for individual blocks, just bundles. 

Are you saying that if I asked for 50..100 and the existing bundles are 25..75 and 75..125, I would get those two bundles back? or nothing, since my requested range didn't encompass any _full_ bundle?

Reply:
maybe we should rope in @MujkicA here. it shouldn't return anything.

Reply:
I've updated the issue on the committer side, probably better if we move this discussion there:

https://github.com/FuelLabs/fuel-block-committer/issues/110